core/primitives/src/stateless_validation.rs

```diff
@@ -24,6 +24,24 @@ pub struct ChunkStateWitness {
     pub signature: Signature,
 }
 
+/// Represents an acknowledgement sent from the chunk producer upon receiving the state witness to
+/// the originator of the witness (chunk producer). This message is currently used for computing
+/// the network round-trip time of sending the state witness to the chunk producer and receiving the
+/// endorsement message. Note that the endorsement message is sent to the next block producer,
+/// while this message is sent back to the originator of the state witness, though this allows
+/// us to approximate the time for transmitting the state witness + transmitting the endorsement.
+#[derive(Debug, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
+pub struct ChunkStateWitnessAck {
```

Comment:
You used a new dedicated message for responding with and ack. Perhaps the networking layer already provides a mechanism for sending a message and waiting for a response. It's only a guess and it's not necessarily a better approach but it could be an alternative.